Sources/MapboxCoreNavigation/Directions.swift

```diff
@@ -0,0 +1,76 @@
+
+import Foundation
+import Network
+import MapboxDirections
+import MapboxNavigationNative
+
+extension Directions {
+    
+    /**
+     Begins asynchronously calculating routes using the given options and delivers the results to a closure.
+     
+     This method retrieves the routes asynchronously from the [Mapbox Directions API](https://www.mapbox.com/api-documentation/navigation/#directions) over a network connection. If a  server error occurs, details about the error are passed into the given completion handler in lieu of the routes. If network error is encountered, onboard routing engine will attempt to provide directions using existing cached tiles.
+     
+     Routes may be displayed atop a [Mapbox map](https://www.mapbox.com/maps/).
+     
+     - parameter options: A `RouteOptions` object specifying the requirements for the resulting routes.
+     - parameter completionHandler: The closure (block) to call with the resulting routes. This closure is executed on the application’s main thread.
+     - returns: The data task used to perform the HTTP request. If, while waiting for the completion handler to execute, you no longer want the resulting routes, cancel this task.
+     */
+    @discardableResult open func calculateWithCache(_ options: RouteOptions, completionHandler: @escaping RouteCompletionHandler) -> URLSessionDataTask? {
```

Comment:
The “withCache” means the method no longer reads like a sentence, so the `options` argument should have a label:

```swift
@discardableResult open func calculateWithCache(options: RouteOptions, completionHandler: @escaping RouteCompletionHandler) -> URLSessionDataTask?
```

Alternatively, we could do something similar to the legacy offline routing functionality in v1._x_, in which the method can either use or not use the cache based on an optional argument:

https://github.com/mapbox/mapbox-navigation-ios/blob/65eb7a1eab9f95234cec713bcfaaad1657e3ab10/Sources/MapboxCoreNavigation/OfflineDirections.swift#L191

In that case, the method signature would look like:

```swift
@discardableResult open func calculate(_ options: RouteOptions, usingCache usesCache: Bool = true, completionHandler: @escaping RouteCompletionHandler) -> URLSessionDataTask?
```

Reply:
I think the "label" approach is preferable because there are 3 scenarios: use online routing only, use online and fallback to offline, use offline only. Thus controlling it with boolean flags is uncomfortable unless we introduce an enum.